src/components/action-bar/action-bar.tsx

```diff
@@ -0,0 +1,204 @@
+ * but its primary goal remains consistent—to
+ * empower users to interact with the software and perform desired actions effortlessly.
+ * @exampleComponent limel-example-action-bar
+ * @exampleComponent limel-example-action-bar-overflow-menu
+ * @exampleComponent limel-example-action-bar-colors
+ * @exampleComponent limel-example-action-bar-floating
+ * @exampleComponent limel-example-action-bar-styling
+ * @exampleComponent limel-example-action-bar-as-primary-component
+ * @private
+ */
+@Component({
+    tag: 'limel-action-bar',
+    shadow: true,
+    styleUrl: 'action-bar.scss',
+})
+export class ActionBar {
+    /**
+     * Items that are placed in the action bar.
+     * These represent primary actions.
+     */
+    @Prop()
+    public actions: Array<ActionBarItem | ListSeparator> = [];
+
+    /**
+     * A label used to describe the purpose of the element to users
+     * of assistive technologies, like screen readers.
+     * Example value: "toolbar"
+     */
+    @Prop({ reflect: true })
+    public accessibleLabel?: string;
+
+    /**
+     * - When set to `fullWidth`, the component will take the
+     * entire width of its container.
+     * - When set to `floating`, the component will get basic stylings
+     * to visualize the floating state.
+     * :::note
+     * You should still properly position the component
+     * according to the structure of your user interface.
+     * For example, use an `absolute` or `fixed` position.
+     * :::
+     */
+    @Prop({ reflect: true })
+    public layout?: 'fullWidth' | 'floating';
+
+    /**
+     * Defines the location that the content of the overflow menu
+     * appears, in relation to its trigger.
+     */
+    @Prop({ reflect: true })
+    public openDirection: OpenDirection;
+
+    /**
+     * Fired when a action bar item has been clicked.
+     */
+    @Event()
+    public itemSelected: EventEmitter<ActionBarItem>;
+
+    @Element()
+    private host: HTMLElement;
+
+    @State()
+    private overflowCutoff: number = this.actions.length;
+
+    private intersectionObserver: IntersectionObserver;
+    private firstRender = true;
+
+    public render() {
+        let overflowActions: Array<MenuItem | ListSeparator> = [];
+        if (this.actions.length) {
+            overflowActions = this.actions.slice(this.overflowCutoff);
+        }
+
+        return (
+            <Host
+                aria-label={this.accessibleLabel}
+                class={{
+                    'is-full-width': this.layout === 'fullWidth',
+                    'is-floating': this.layout === 'floating',
+                }}
+            >
+                <div class="items">
+                    {this.actions.map(this.renderActionBarItem)}
+                </div>
+                {this.renderOverflowMenu(overflowActions)}
+            </Host>
+        );
+    }
+
+    public componentDidLoad() {
+        this.createIntersectionObserver();
+    }
```

Comment:
This doesn't create a new intersection observer after disconnect (the same problem that mentioned in https://github.com/Lundalogik/lime-elements/pull/2315#discussion_r1214224949), but what's more important, it doesn't account for any changed actions bar items after first render.

Reply:
A suggestion is to do something in `componentDidRender()` instead, and only create a new intersection observer (or maybe just update which elements it observes, if that's possible) if the inputs changed.

So you could compare the result of `this.host.shadowRoot.querySelectorAll('limel-action-bar-item')` to see if they are the same as the observed elements or not.